src/KafkaSchemaRegistryApiApiClient.php

```diff
@@ -0,0 +1,229 @@
+     */
+    private $httpClient;
+
+    /**
+     * KafkaSchemaRegistryApi constructor.
+     * @param HttpClientInterface $httpClient
+     */
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * @return array
+     */
+    public function getSubjects(): array
+    {
+        return $this->httpClient->call('GET', 'subjects') ?? [];
+    }
+
+    /**
+     * @param string $subjectName
+     * @return array
+     */
+    public function getAllSubjectVersions(string $subjectName): array
+    {
+        return $this->httpClient->call('GET', sprintf('subjects/%s/versions', $subjectName)) ?? [];
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $version
+     * @return string
+     */
+    public function getSchemaByVersion(string $subjectName, string $version = self::VERSION_LATEST): string
+    {
+        return $this
+                ->httpClient
+                ->call('GET', sprintf('subjects/%s/versions/%s', $subjectName, $version))['schema'];
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $version
+     * @return bool
+     */
+    public function deleteSchemaVersion(string $subjectName, string $version = self::VERSION_LATEST): bool
+    {
+        $this->httpClient->call('DELETE', sprintf('subjects/%s/versions/%s', $subjectName, $version));
+        return true;
+    }
+
+    /**
+     * @param int $id
+     * @return string
+     */
+    public function getSchemaById(int $id): string
+    {
+        return $this->httpClient->call('GET', sprintf('schemas/ids/%s', $id))['schema'];
+    }
+
+    /**
+     * @param string $schema
+     * @param string $subjectName
+     * @return array
+     */
+    public function registerNewSchemaVersion(string $subjectName, string $schema): array
+    {
+        return $this
+                ->httpClient
+                ->call(
+                    'POST',
+                    sprintf('subjects/%s/versions', $subjectName),
+                    $this->createRequestBodyFromSchema($schema)
+                ) ?? [];
+    }
+
+    /**
+     * @param string $schema
+     * @param string $subjectName
+     * @param string $version
+     * @return bool
+     * @throws Exception
+     */
+    public function checkSchemaCompatibilityForVersion(
+        string $subjectName,
+        string $schema,
+        string $version = self::VERSION_LATEST
+    ): bool {
+        try {
+            $results = $this
+                   ->httpClient
+                   ->call(
+                       'POST',
+                       sprintf('compatibility/subjects/%s/versions/%s', $subjectName, $version),
+                       $this->createRequestBodyFromSchema($schema)
+                   ) ?? [];
+        } catch (SubjectNotFoundException $e) {
+            return true;
+        }
+
+        /** @var bool $result */
+        $result = $results['is_compatible'];
+        return $result;
```

Comment:
```suggestion
        return $results['is_compatible'] === true;
```
this should do as well, i guess you reassigned it because of phpstan

Reply:
Yes, because of PHPStan it's like that...

Reply:
changed